src/logic/calculate.test.js

```diff
@@ -0,0 +1,18 @@
+import calculate from './calculate';
+
+test('should first', () => {
+  expect(
+    calculate(
+      {
+        total: null,
+        next: null,
+        operation: null,
+      },
+      'AC',
+    ),
+  ).toMatchObject({
+    total: null,
+    next: null,
+    operation: null,
+  });
+});
```

Comment:
- [ ] It is important to test all functionalities to ensure there is no bug in your code. Please add more tests for calculate.js, ensure it has individual for tests all the functionalities and scenarios.